### PR TITLE
Welling/soft assay types start

### PR DESCRIPTION
**Pull request by jswelling**
This PR:
* Adds a fallback attempt to access entity-api with no token for a uuid for which the current token has failed.  This will support calls for published datasets from other groups.
* Adds an 'is_support' field to vitessce_hints for datasets which need to be vis-lifted
* Corrects an incorrect 'primary' attribute for SEQFish_pyramid in the saved copy of assay_types.yaml
* Adds a uuid for a (currently) unpublished dataset to test_examples
* Makes code for dealing with derived datasets with no data_types info
* Updates the rule set with these changes

### CODE REVIEW

File: src/routes/assayclassifier/__init__.py [hunk 1]
```diff
@@ -62,12 +62,20 @@ def get_ds_assaytype(ds_uuid: str):
         entity_api_url = current_app.config['ENTITY_WEBSERVICE_URL']
         groups_token = groups_token_from_request_headers(request.headers)
```
Comment (jpuerto-psc) [inline src/routes/assayclassifier/__init__.py:63]: @jswelling - does this work if the Authorization header is missing?

File: src/routes/assayclassifier/__init__.py [hunk 2]
```diff
@@ -62,12 +62,20 @@ def get_ds_assaytype(ds_uuid: str):
...
+            entity_api = EntitySdk(service_url=entity_api_url)
+            entity = entity_api.get_entity_by_id(ds_uuid) # may again raise SDKException
         if 'metadata' in entity.ingest_metadata:
             metadata = entity.ingest_metadata['metadata']
         else:
-            metadata = {'entity_type': entity.entity_type,
-                        'data_types': entity.data_types}
+            if hasattr(entity, 'data_types') and entity.data_types:
+                metadata = {'entity_type': entity.entity_type,
+                            'data_types': entity.data_types}
```
Comment (yuanzhou) [inline src/routes/assayclassifier/__init__.py:75]: @jswelling @jpuerto-psc just so we are on the same page, this `data_types` will go away eventually and the new `dataset_type` is replacing it.

File: src/routes/assayclassifier/assay_types.yaml [hunk 3]
```diff
@@ -416,7 +416,7 @@ seqFish:
 seqFish_pyramid:
     description: seqFISH [Image Pyramid]
     alt-names: []
-    primary: true
+    primary: false
```
Comment (yuanzhou) [inline src/routes/assayclassifier/assay_types.yaml:419]: @jswelling according to @jpuerto-psc 's comment from the previous PR: https://github.com/hubmapconsortium/ingest-api/pull/441#discussion_r1425427802

Should we keep it as `primary: true` to be the same as the value specified in the deprecated yaml https://github.com/hubmapconsortium/search-api/blob/main/src/search-schema/data/definitions/enums/assay_types.yaml#L419?

**Review by yuanzhou — COMMENTED** [review]
@jswelling in addition to the comments below, should I go ahead and make this `GET /assaytype/<dataset-id>` call public accessible without using a token in the Gateway?

File: src/routes/assayclassifier/assay_types.yaml (hunk 3 above)
Comment (jpuerto-psc) [reply src/routes/assayclassifier/assay_types.yaml:419]: No, that value was wrong @yuanzhou - this should be set to false.

**Comment by jpuerto-psc** [comment]
@yuanzhou - yes this endpoint should be public accessible without a token. The POST endpoint `POST /assaytype` should also be made public accessible without a token. My comment RE whether this works without an Auth header should not hold us from releasing these changes.

File: src/routes/assayclassifier/__init__.py (hunk 2 above)
Comment (jpuerto-psc) [reply src/routes/assayclassifier/__init__.py:75]: @yuanzhou - believe we're all on the same page here, let's go ahead and release this for now as Joel is not in until next Th and this should not block a release to dev-integrate

File: src/routes/assayclassifier/__init__.py (hunk 1 above)
Comment (jpuerto-psc) [reply src/routes/assayclassifier/__init__.py:63]: Confirmed - this is working as expected.

**Review by yuanzhou — APPROVED** [review]

**Pull request merged**